assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,218 @@
+
+  const sequencedVideoOptimization = useCallback(() => {
+    // only attempt to optimize videos that are not currently being transcoded
+    const optimizingResources = unoptimizedVideos.filter(
+      ({ resource }) => !resource?.isTranscoding
+    );
+    return new Promise((res) => {
+      optimizingResources
+        .map(({ resource }) => async () => {
+          await optimizeVideo({ resource }).finally(() => {
+            optimizingResourcesRef.current[resource.id] = STATES.UPLOADED;
+          });
+        })
+        .reduce((p, fn) => p.then(fn), Promise.resolve())
+        .then(() => {
+          res();
+        });
+    });
+  }, [unoptimizedVideos, optimizeVideo]);
+
+  const handleUpdateVideos = useCallback(async () => {
+    unoptimizedVideos.forEach(({ resource }) => {
+      if (!resource.isTranscoding) {
+        optimizingResourcesRef.current[resource.id] = STATES.UPLOADING;
+      }
+    });
+    await sequencedVideoOptimization();
+  }, [sequencedVideoOptimization, unoptimizedVideos]);
+
+  const handleClickThumbnail = useCallback(
+    (element) => async () => {
+      setHighlights({
+        pageId: element.pageId,
+        elementId: element.id,
+      });
+      if (
+        !element.resource?.isTranscoding &&
+        ![STATES.UPLOADING].includes(
+          optimizingResourcesRef.current[element.resource.id]
+        )
+      ) {
+        optimizingResourcesRef.current[element.resource.id] = STATES.UPLOADING;
+        await optimizeVideo({ resource: element.resource }).finally(() => {
+          optimizingResourcesRef.current[element.resource.id] = STATES.UPLOADED;
+        });
+      }
+    },
+    [optimizeVideo, setHighlights]
+  );
+
+  const { footer, title } = PRIORITY_COPY.videoNotOptimized;
+
+  const isTranscoding = useMemo(
+    () =>
+      Object.values(optimizingResourcesRef.current).includes(
+        STATES.UPLOADING
+      ) || unoptimizedVideos.some((video) => video.resource?.isTranscoding),
+    [unoptimizedVideos]
+  );
+
+  const isRendered = unoptimizedVideos.length > 0;
+  useRegisterCheck('VideoOptimization', isRendered);
+
+  return (
+    isRendered && (
+      <ChecklistCard
+        cta={
+          (enableBulkVideoOptimization || unoptimizedVideos.length === 1) && (
+            <OptimizeButton
+              type={BUTTON_TYPES.SECONDARY}
+              size={BUTTON_SIZES.SMALL}
+              onClick={handleUpdateVideos}
+              disabled={isTranscoding}
+            >
+              {/* todo copy should count the number of videos */}
+              {isTranscoding
+                ? _n(
+                    'Optimizing video',
+                    'Optimizing videos',
+                    unoptimizedVideos.length,
+                    'web-stories'
+                  )
+                : _n(
+                    'Optimize video',
+                    'Optimize all videos',
+                    unoptimizedVideos.length,
+                    'web-stories'
+                  )}
```

Comment:
That's not how `_n()` works.

If you use `_n()`, you **must** include the actual count in the string. Example: `_n('Optimize %s video', 'Optimize %s videos'`.

If we don't want  to include the count, then we **must** do: `(unoptimizedVideos.length === 1) ? 'Optimize video' : 'Optimize all videos'`